corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -194,44 +219,167 @@ def _rebuild_queues(self, pool):
+                #           "create": {"case_name": "Abigail",
+                #                      "case_type": "case",
+                #                      "owner_id": "7ea59f550f35758447400937f800f78c"}},
+                #  "meta": {"@xmlns": "http://openrosa.org/jr/xforms",
+                #           "appVersion": {"#text": "Formplayer Version: 2.43",
+                #                          "@xmlns": "http://commcarehq.org/xforms"},
+                #           "deviceID": "Formplayer",
+                #           "drift": "",
+                #           "instanceID": "4378faa0-58b1-4c51-9310-fef1cda29707",
+                #           "timeEnd": datetime.datetime(2019, 2, 7, 9, 15, 48, 575000),
+                #           "timeStart": datetime.datetime(2019, 2, 7, 9, 15, 35, 853000),
+                #           "userID": "7ea59f550f35758447400937f800f78c",
+                #           "username": "admin@example.com"},
+                #  "name": "Abigail"}
+                if 'owner_id' in case['create'] and case['create']['owner_id'] in self._id_map:
+                    case['create']['owner_id'] = self._id_map[case['create']['owner_id']]
+
+            if 'update' in case:
+                # {"#type": "data",
+                #  "@name": "Followup Form",
+                #  "@uiVersion": "1",
+                #  "@version": "7",
+                #  "@xmlns": "http://openrosa.org/formdesigner/5942B189-B697-4139-898D-79C87428F83A",
+                #  "case": {"@case_id": "567d96f2-6599-49db-9bf7-aa52824d0ab6",
+                #           "@date_modified": datetime.datetime(2019, 2, 12, 12, 57, 10, 843000),
+                #           "@user_id": "f931c5dc7c40499ebbe5141803579d67",
+                #           "@xmlns": "http://commcarehq.org/case/transaction/v2",
+                #           "update": {"case_name": "Andrea"}},
+                #  "meta": { … },
+                #  "name": "Andrea"}
+                pass
+
+        meta = couch_form.form['meta']
+        if 'userID' in meta and meta['userID'] in self._id_map:
+            meta['userID'] = self._id_map[meta['userID']]
+
+        # TODO:
+        #     - [ ] Basic forms:
+        #           - [✔] Change IDs in case create
+        #           - [✔] Change IDs in case update
+        #           - [ ] Change IDs in child cases
+        #           - [ ] Change IDs in survey module forms
+        #     - [ ] Advanced forms:
+        #           - [ ] Change IDs in multiple case creates
+        #           - [ ] Change IDs in multiple case updates
+        #           - [ ] Change IDs in child cases
+        #     - [ ] System forms (imported cases)
+        #     - [ ] Change IDs in save-to-case
+        #     - [ ] Questions that store location IDs or user IDs
+
+        return couch_form
+
+    def _map_case_ids(self, couch_case):
+        if self.src_domain == self.dst_domain:
+            return couch_case
+
+        # TODO: Is this ever necessary?
```

Comment:
what do you mean?

Reply:
I should drop that comment. `_map_case_ids()` gets called by `_copy_unprocessed_case()`, so the user/location IDs in the case haven't already been mapped by the forms that created/updated it. It's necessary every time it's called.